### PR TITLE
refactor: Kotlin migration of AbstractFlashcardViewerCommandTest

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Done to test https://github.com/ankidroid/Anki-Android/pull/10779 in test

### CODE REVIEW

File: AnkiDroid/kotlinMigration.gradle [hunk 1]
```diff
@@ -42,8 +42,8 @@ permission notice:
 
 // Example of class name: "/com/ichi2/anki/UIUtils.kt"
 // Ensure that it starts with '/' (slash)
-def source = Source.MAIN
-def className = ""
+def source = Source.TEST
+def className = "/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt"
```
Comment (david-allison) [inline AnkiDroid/kotlinMigration.gradle]: This is incorrect. Should be `/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt`

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 2]
```diff
@@ -13,233 +13,181 @@
...
-
-        viewer.executeCommand(ViewerCommand.COMMAND_TOGGLE_FLAG_RED);
-        viewer.executeCommand(ViewerCommand.COMMAND_TOGGLE_FLAG_RED);
-
-        assertThat(viewer.getLastFlag(), is(FLAG_NONE));
+    fun doubleTapSetsNone() {
+        val viewer = viewer
+        viewer.executeCommand(ViewerCommand.COMMAND_TOGGLE_FLAG_RED)
+        viewer.executeCommand(ViewerCommand.COMMAND_TOGGLE_FLAG_RED)
+        MatcherAssert.assertThat(viewer.lastFlag, Matchers.`is`(CardMarker.FLAG_NONE))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:39]: fix: re-import `MatcherAssert` and `Matchers`

cleanup: replace `is`

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 3]
```diff
@@ -13,233 +13,181 @@
...
     @Test
-    public void doubleTapSetsNone() {
-        CommandTestCardViewer viewer = getViewer();
-
-        viewer.executeCommand(ViewerCommand.COMMAND_TOGGLE_FLAG_RED);
-        viewer.executeCommand(ViewerCommand.COMMAND_TOGGLE_FLAG_RED);
-
-        assertThat(viewer.getLastFlag(), is(FLAG_NONE));
+    fun doubleTapSetsNone() {
+        val viewer = viewer
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:36]: cleanup (for class): This makes a lot less sense.

`val viewer = viewer` makes a lot less sense than `CommandTestCardViewer viewer = getViewer();`. needs renames

cleanup: rename `getViewer -> createViewer`

File: AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt [hunk 4]
```diff
@@ -13,233 +13,181 @@
...
+        MatcherAssert.assertThat(command.toString(), viewer.lastFlag, Matchers.`is`(CardMarker.FLAG_NONE))
+    }
+
+    private val viewer: CommandTestCardViewer
+        get() = CommandTestCardViewer(cardWith(CardMarker.FLAG_NONE))
+
+    private fun cardWith(@FlagDef flag: Int): Card {
+        val c = Mockito.mock(Card::class.java)
+        val flags = intArrayOf(flag)
+        Mockito.`when`(c.userFlag()).then { flags[0] }
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerCommandTest.kt:146]: cleanup: use `whenever`

**Review by david-allison — CHANGES_REQUESTED** [review]

**Comment by mikehardy** [comment]
Would be nice if script made the PR title of format: `refactor: Kotlin migration of <filename>` ?

**Pull request closed without merging**